mkinitcpio-install.sh

```diff
@@ -52,6 +52,35 @@ run_command() {
     esac
 }
 
+concatenate_unit() {
+    # concatenate units and their potential drop-in files
+    # we do this manually, as in a chroot `systemctl cat` will not work
+    local unit_name="$1"
+    local unit_content=""
+    local service_path=""
+    local override=""
+    # add the top-most service
+    for service_path in {/usr/lib,/etc,/run}/systemd/system; do
+        if [ -f "${service_path}/${unit_name}" ]; then
+          unit_content=""
+          unit_content+="# ${service_path}/${unit_name}\n"
+          unit_content+="$(cat "${service_path}/${unit_name}")\n\n"
```

Comment:
Just echo/printf directly and `break`

```bash
echo "# ${service_path}/${unit_name}"
cat "${service_path}/${unit_name}"
break
```
`echo` will have a linebreak as well es `cat` since it's not stripped due to subshell. You can add additional newlines in the drop-in loop.

Reply:
True I didn't consider that. I have been doing too much python lately :D